### PR TITLE
[RLlib] Fix type hints for `original_batches` in callbacks.

### PR DESCRIPTION
**Pull request by XuehaiPan**
## Why are these changes needed?

Fix type hints for `original_batches` in `on_postprocess_trajectory`.

## Related issue number

N/A

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/callbacks.py [hunk 1]
```diff
@@ -196,7 +196,7 @@ def on_postprocess_trajectory(
         policy_id: PolicyID,
         policies: Dict[PolicyID, Policy],
         postprocessed_batch: SampleBatch,
-        original_batches: Dict[AgentID, SampleBatch],
+        original_batches: Dict[AgentID, Tuple[Policy, SampleBatch]],
```
Comment (gjoliver) [inline rllib/agents/callbacks.py:199]: what, that's surprising ...
how can we mix these 2 things together?

Comment (gjoliver) [reply rllib/agents/callbacks.py:199]: oh it's Tuple, not Union, I got scared.
Thanks!

**Review by gjoliver — APPROVED** [review]
@sven1977 can you help merge?

File: rllib/agents/callbacks.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/callbacks.py:199]: :) @gjoliver 

I think there is only very few places in RLlib where we mix different types e.g. in a return value (for example in the sampler code `_process_observations()`) and no, we probably shouldn't do this.

**Pull request merged**